addons/product/models/product_template.py

```diff
@@ -602,6 +602,15 @@ def _create_variant_ids(self):
                 variants_to_create += current_variants_to_create
                 variants_to_activate += current_variants_to_activate
 
+            else:
+                ptav_ids = lines_without_no_variants.product_template_value_ids.ids
+                active_existing_variants = {variant.product_template_attribute_value_ids: variant for variant in all_variants if variant.active}
```

Comment:
If I use `_is_variant_possible` it will returns `False` as soon as the product is archived. https://github.com/odoo/odoo/blob/df12c85816dfa59f298c3e6a077bd88e4b7aa6a6/addons/product/models/product_template.py#L820. So, I guess I should do something else than use `_is_variant_possible` or is there a workaround?

Reply:
I forgot about that part, that's annoying.en we'll need to get the first few conditions from `_is_combination_possible`.

Reply:
To be more precise the `product_template_attribute_value_ids` from the variant must contain exactly one value for each `valid_product_template_attribute_line_ids`, and each value must be part of the possible active values of the line.

That's the first 3 conditions on `_is_combination_possible`. Maybe they should be factored to a new method to avoid copy/paste. Those correspond to the "verify config on the template" part.
The following conditions of `_is_combination_possible` are "verify variant state" then "verify exclusion rules", and those aren't needed to know if the variant should be created/activated or not.

Reply:
\+ there will be a tricky part because `product_template_attribute_value_ids` on variant don't contain "no variant" attributes, except the check for `_is_combination_possible` is expecting them to be given as well.

So factoring the conditions might not even work because "no variant" attributes must be excluded from the check in our case. Now that I think about it even the current `_is_variant_possible` might actually incorrectly return false because of this, as per note
`This will always exclude variants for templates that have `no_variant` because the variant itself will not be the full combination.`

I think it was previously never the case that `_is_variant_possible` was called in such case, but in our case, it can and will happen as soon as there is such config (dynamic attribute + no variant attribute on the same template).

Reply:
I pushed new changes, please let me know what you think about it. The test seems to pass correctly.

Reply:
In case you miss it, I just changed
```py
if self.valid_product_template_attribute_line_ids.product_template_value_ids._only_active() < combination:
```

into

```py
valid_active_values_ids = self.valid_product_template_attribute_line_ids.product_template_value_ids._only_active().ids
if not all([id in valid_active_values_ids for id in combination.ids]):
```

because previously, when comparing product.attribute.value(1,) and product.attribute.value(2,) it will pass but I don't think it should pass because attribute 1 is not available.